cmake/Runtime.cmake

```diff
@@ -6,20 +6,112 @@
-list(INSERT LD_LIBRARY_PATH 0 ${CMAKE_BINARY_DIR}/lib)
+get_property(FAIRROOT_RUNTIME GLOBAL PROPERTY FAIRROOT_RUNTIME)
+
+IF(CMAKE_SYSTEM_NAME MATCHES Linux)
+  IF(FAIRROOTPATH)
+    configure_file(${FAIRROOTPATH}/share/fairroot/cmake/scripts/check_system.sh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.sh
+                  )
+    configure_file(${FAIRROOTPATH}/share/fairroot/cmake/scripts/check_system.csh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.csh
+                  )
+  ELSE(FAIRROOTPATH)
+    configure_file(${PROJECT_SOURCE_DIR}/cmake/scripts/check_system.sh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.sh
+                  )
+    configure_file(${PROJECT_SOURCE_DIR}/cmake/scripts/check_system.csh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.csh
+                  )
+  ENDIF(FAIRROOTPATH)
+  
+  EXECUTE_PROCESS(COMMAND lsb_release -sd 
+                   OUTPUT_VARIABLE _linux_flavour
+                   OUTPUT_STRIP_TRAILING_WHITESPACE
+                  )
+
+  IF(_linux_flavour)
+    STRING(REGEX REPLACE "^\"" "" _linux_flavour ${_linux_flavour})
+    STRING(REGEX REPLACE "\"$" "" _linux_flavour ${_linux_flavour})
+  ENDIF(_linux_flavour)
+
+  EXECUTE_PROCESS(COMMAND uname -m 
+                  OUTPUT_VARIABLE _system 
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                 )
+ 
+ElseIf(CMAKE_SYSTEM_NAME MATCHES Darwin)
+  IF(FAIRROOTPATH)
+    configure_file(${FAIRROOTPATH}/share/fairroot/cmake/scripts/check_system_mac.sh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.sh
+                  )
+    configure_file(${FAIRROOTPATH}/share/fairroot/cmake/scripts/check_system_mac.csh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.csh
+                  )
+  ELSE(FAIRROOTPATH)
+    configure_file(${PROJECT_SOURCE_DIR}/cmake/scripts/check_system_mac.sh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.sh
+                  )
+    configure_file(${PROJECT_SOURCE_DIR}/cmake/scripts/check_system_mac.csh.in
+                   ${CMAKE_CURRENT_BINARY_DIR}/check_system.csh
+                  )
+  ENDIF(FAIRROOTPATH)
+  EXECUTE_PROCESS(COMMAND uname -sr 
+                  OUTPUT_VARIABLE _linux_flavour
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                 )
+  EXECUTE_PROCESS(COMMAND uname -m 
+                  OUTPUT_VARIABLE _system 
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                 )
+ENDIF(CMAKE_SYSTEM_NAME MATCHES Linux)
+
+set(_INSTALLDIR ${CMAKE_BINARY_DIR})  
+set(_BINDIR ${CMAKE_BINARY_DIR})  
+
+set(VMCWORKDIR ${CMAKE_SOURCE_DIR}/examples) # ???
+
+set(FAIRLIBDIR ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR})  
+
+set(USE_VGM 1)
+
+list(APPEND LD_LIBRARY_PATH ${FAIRLIBDIR}) 
 list(REMOVE_DUPLICATES LD_LIBRARY_PATH)
-set(VMCWORKDIR ${CMAKE_SOURCE_DIR}/examples)
+set(MY_LD_LIBRARY_PATH ${LD_LIBRARY_PATH})
```

Comment:
What about Mac and DYLD_LIBRARY_PATH?

Reply:
The content of `MY_LD_LIBRARY_PATH` gets emitted in the `FairRootConfig.sh` script as `LD_LIBRARY_PATH` and `DYLD_LIBRARY_PATH`.